### ISSUE TITLE
wallwalking issues

### ISSUE THREAD

**Issue by overvolting** [Unvanquished/Unvanquished#609]
Trying to wall walk around sharp bends is difficult and you fall off. Trying to do it around circular holes is nigh impossible. For example, parpax hanger hole or nano hole.

Also, even when i turn off auto pitch, it still auto pitches and spins me around.

**Comment by Kangz** [comment]
The wallwalking physics is whre there is the following comment (or something close):
    // beware the evil mindfuck from hell
As much as we'd like to have an awesome wallwalking experience I fear this will have to wait for someone to change the physics code deeply.

**Comment by Viech** [comment]
I don't think this needs a new physics backend. Wallwalking is merely the combination of shared movement code that sets your orientation, attachment and the like and client side interpolation of the first person view direction. The existing code is horrible to read or maintain but I think someone could just nuke and rewrite it on top of the current physics. I don't think we need to preserve anything about the current wallwalking mechanics, especially not about first person visualization and controls. (Given how detailed some of our walls and ceilings are, I don't think the first person perspective should adjust to the surface normal.)

**Comment by illwieckz** [comment]
One suggestion I did long time ago (I believe in the forums but I don't find it) is to start to move the player view before the wall, by tracing.

Basically (imagine `///o` is a granger):


```
                     wall
                    /
granger            /
       ///o       /
floor ___________/ ← angle between floor and wall
      |    |    |
      0%   50%  100% ← interpolation ratio
```

The idea is to trace in front of the player, if a wall is within a given range, apply to the player view a fraction of the wall angle to the player view, the closer the player is to the wall, the more the angle of the player view matches the angle of the wall.

Of course this does not solve all problems, for example this doesn't take in account walls when side walking that would be as abrupt as before. But we can start experimenting that before thinking about interpolating in more than one direction…
